docs/docs/install/docker/index.md

```diff
@@ -0,0 +1,146 @@
+     listening on port `8000` and additionally
+     - Passes configuration options as environment variables to the docker container. Refer to [Configuring Mathesar webserver](../configuration.md#backend-configuration) for setting the correct value to these configuration options and for additional configuration options. The configuration options used in the above command are
+       - DJANGO_DATABASE_URL
+       - DJANGO_DATABASE_KEY
+       - MATHESAR_DATABASES
+       - SECRET_KEY
+       - Creates two [named docker volumes](https://docs.docker.com/storage/volumes/)
+          ```
+               - static # For storing static assets like CSS, js files
+               - media # For storing user-uploaded media files
+          ```
+       - Sets the container name as `mathesar_service` using the `--name` parameter, runs the container in a detached mode using the `--detach` parameter, and exposes the port `8000` to the `localhost`. Refer to [Docker documentation](https://docs.docker.com/engine/reference/commandline/run/#options) for additional configuration options.
+       - Finally, sets the container entrypoint to run the script `run.sh` which starts a gunicorn server on port `8000`
+
+2. Verify if the Mathesar Server is running successfully:
+    ```bash
+    docker logs -f mathesar_service
+    ```
+
+3. Run Mathesar Installation script
+    ```bash
+    # Install Mathesar types and casting function on the databases specified using the `MATHESAR_DATABASES` env variable in the previous step, run database migrations on the Django meta database, and store the static files in the `static` volumes.
+    docker exec mathesar_service python install.py --skip-confirm
+    ```
+
+4. Create a superuser
+    ```bash
+    # Run the Django `createsuperuser` command. Refer https://docs.djangoproject.com/en/4.2/ref/django-admin/#createsuperuser
+    docker exec -it mathesar_service python manage.py createsuperuser
+    ```
+    A prompt will appear to ask for the superuser details, fill in the details to create a superuser. At least one superuser is necessary for accessing Mathesar.
+
+## Frequently Asked Questions
+
+#### Upgrading from a previous version
+
+1. Stop your existing Mathesar container:
+
+    ```bash
+    docker stop mathesar_service
+    ```
+
+2. Bump the image version in the `docker run` command you usually use to run your
+   Mathesar and start up a brand-new container:
+
+    ```bash
+    
+        docker run \
+          -d \
+          # We haven't yet deleted the old Mathesar container so you need to start this new one
+          # with a different name to prevent an error like:
+          # `response from daemon: Conflict. The container name "/mathesar_service" is already in use` 
+          --name mathesar_service_version_REPLACE_WITH_NEW_VERSION \
+          # YOUR STANDARD ARGS HERE
+          mathesar/mathesar-prod:REPLACE_WITH_LATEST_VERSION
+    ```
```

Comment:
Wouldn't stopping the service, pulling the latest image, and starting it again, ensure that the latest image is used?

Why would the user have to manually create a new container with a different name?

Reply:
No docker won't do it automatically. If the same container name is used for a different image, it throws up the `Conflict. The container name "/mathesar_service" is already in use` which I have mentioned in the comment